### ISSUE TITLE
Support date/datetime values as sort keys and in CQL comparison predicates

### ISSUE THREAD

**Issue by cportele** [interactive-instruments/ldproxy#480]
Note: Only for SQL feature providers.

Should be implemented together with #335.

**Comment by azahnen** [comment]
I pushed a commit in interactive-instruments/xtraplatform-spatial#84 to support date/datetime as secondary sort keys. Note that using them as primary sort keys in the provider will lead to unexpected results as date/datetime values will not be unique in most cases.

**Comment by cportele** [comment]
@azahnen - I have tested the sorting with the NOTAM API from T17 ([link to the deployed data](https://t17.ldproxy.net/fns/collections/notam/items)), by adding

```yaml
    - buildingBlock: SORTING
      enabled: true
      sortables:
      - issued
```
`sortby=issued` works, but `sortby=-issued` results in no features (but numberMatched is 10).

**Comment by azahnen** [comment]
Is it possible that `issued` contains null values? I tested with geoval observations and `-resultTime.start` worked as expected. I think the code does not expect null values when creating the min/max filters for sort keys, so I would have to fix that.

**Comment by cportele** [comment]
No, there is no `null` value ([query link](https://t17.ldproxy.net/fns/collections/notam/items?filter=issued+is+null&f=json)). 

But, in general, sortby should also work with `null` values, not just for instants.

**Comment by azahnen** [comment]
I agree, but that might have explained it. I will test with t17 tomorrow.

**Comment by azahnen** [comment]
I pushed a fix for the datetime issue, which occurred due to the timezone being ignored. Ascending did not work correctly either, but since the page showed 9 features, it wasn't so obvious.

That being said, after some more tests I came to the conclusion that the current implementation does not support non-unique sort keys. I tried to find an easy fix, but it will require some deeper changes to the query generation, so I created a separate issue #488.

**Comment by azahnen** [comment]
@cportele This should work as expected in `xtraplatform-spatial` version `5.0.0-sql-queries-SNAPSHOT`.

**Comment by cportele** [comment]
@azahnen - I tried with 'gebaeude' and am getting strange results with `sortby=name` and `sortby=-measuredHeight`. I have not yet tried to debug why that happens.
